portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/locale/Locale.testcase

```diff
@@ -467,17 +467,15 @@ definition {
 			pageName = "Language Selector Page",
 			siteName = "Guest");
 
-		Click(
-			key_language = "català (espanya, valencia)",
-			locator1 = "Language#LONG_TEXT_LINK");
+		Click(locator1 = "//a[contains(@class, 'long-text') and @lang='ca-ES-VALENCIA']");
 
 		AssertLocation(value1 = "${portalURL}/ca-ES-VALENCIA/web/guest/language-selector-page");
 
 		AssertTextEquals(
 			locator1 = "UserBar#USER_SIGN_IN",
 			value1 = "Accedeix");
 
-		Click(locator1 = "//a[contains(@class, 'long-text') and contains(., 'српски (србија, latin)')]");
+		Click(locator1 = "//a[contains(@class, 'long-text') and @lang='sr-RS-latin']");
```

Comment:
@tinatian I also changed this to do the same thing to prevent future issue.
This can be a larger task to improve functional test compatibility that we can do in the future.